### PR TITLE
Service update

### PR DESCRIPTION
**Pull request by nmorgan-cb**

### CODE REVIEW

File: src/main/java/com/coinbase/core/client/CoinbaseNetHttpClient.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.List;
+import java.util.Map;
+
+public abstract class CoinbaseNetHttpClient implements CoinbaseClient {
+    protected static final ObjectMapper mapper = new ObjectMapper();
+    private final HttpClient client;
+    private final CoinbaseCredentials credentials;
+    private final String baseUrl;
+
+    public CoinbaseNetHttpClient(CoinbaseCredentials credentials, String baseUrl) {
```
Comment (rcbgr) [inline src/main/java/com/coinbase/core/client/CoinbaseNetHttpClient.java:26]: Add a new constructor that doesn't take baseUrl or http client


public CoinbaseNetHttpClient(CoinbaseCredentials credentials)

Comment (rcbgr) [reply src/main/java/com/coinbase/core/client/CoinbaseNetHttpClient.java:26]: Use the default http client if not passed

File: src/main/java/com/coinbase/core/client/CoinbaseNetHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            JsonNode jsonNode = mapper.readTree(jsonString);
+
+            List<String> queryParameters = new ArrayList<>();
+            jsonNode.fields().forEachRemaining(entry -> {
+                String key = URLEncoder.encode(entry.getKey(), StandardCharsets.UTF_8);
+                JsonNode valueNode = entry.getValue();
+                if (valueNode.isArray()) {
+                    valueNode.forEach(arrayElement -> {
+                        String value = URLEncoder.encode(arrayElement.asText(), StandardCharsets.UTF_8);
+                        queryParameters.add(key + "=" + value);
```
Comment (rcbgr) [inline src/main/java/com/coinbase/core/client/CoinbaseNetHttpClient.java:106]: Use "%s: %s".format(key, value);

**Review by rcbgr — CHANGES_REQUESTED** [review]

**Review by rcbgr — APPROVED** [review]
LGTM. Would like to see some tests

**Review by jc-cb — APPROVED** [review]

**Pull request merged**